saleor/static/dashboard-next/products/components/ProductVariants/ProductVariants.tsx

```diff
@@ -0,0 +1,174 @@
+    height: theme.spacing.unit,
+    marginRight: theme.spacing.unit,
+    width: theme.spacing.unit
+  };
+  return {
+    alignRightText: {
+      textAlign: "right" as "right"
+    },
+    denseTable: {
+      "& td, & th": {
+        paddingRight: theme.spacing.unit * 3
+      }
+    },
+    greenDot: {
+      ...dot,
+      backgroundColor: green[500]
+    },
+    link: {
+      color: blue[500],
+      cursor: "pointer"
+    },
+    redDot: {
+      ...dot,
+      backgroundColor: red[500]
+    }
+  };
+});
+
+export const ProductVariants = decorate<ProductVariantsProps>(
+  ({
+    classes,
+    disabled,
+    variants,
+    fallbackPrice,
+    onRowClick,
+    onVariantAdd
+  }) => (
+      <Card>
+        <PageHeader title={i18n.t("Variants")}>
+          {!!onVariantAdd && (
+            <IconButton disabled={disabled} onClick={onVariantAdd}>
+              <AddIcon />
+            </IconButton>
+          )}
+        </PageHeader>
+        <Table className={classes.denseTable}>
+          <TableHead>
+            <TableRow>
+              <Hidden smDown>
+                <TableCell>{i18n.t("SKU")}</TableCell>
+              </Hidden>
+              <TableCell>{i18n.t("Name")}</TableCell>
+              <TableCell>{i18n.t("Status")}</TableCell>
+              <Hidden smDown>
+                <TableCell className={classes.alignRightText}>
+                  {i18n.t("Price")}
+                </TableCell>
+              </Hidden>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {variants === undefined || variants === null || fallbackPrice === undefined ? (
+              <TableRow>
+                <Hidden smDown>
+                  <TableCell>
+                    <Skeleton />
+                  </TableCell>
+                </Hidden>
+                <TableCell>
+                  <Skeleton />
+                </TableCell>
+                <TableCell>
+                  <Skeleton />
+                </TableCell>
+                <Hidden smDown>
+                  <TableCell className={classes.alignRightText}>
+                    <Skeleton />
+                  </TableCell>
+                </Hidden>
+              </TableRow>
+            ) : variants.length > 0 ? (
+              variants.map(variant => {
+                const price = variant.priceOverride || fallbackPrice;
+                return (
+                  <TableRow key={variant.id}>
+                    <Hidden smDown>
+                      <TableCell>{variant.sku}</TableCell>
+                    </Hidden>
+                    <TableCell className={
+                      onRowClick ? classes.link : ""}
+                      onClick={() => {
+                        if (onRowClick) {
+                          onRowClick(variant.id);
+                        }
+                      }}>
+                      {variant.name}
+                    </TableCell>
+                    <TableCell>
+                      <span
```

Comment:
I think this could become a component. Otherwise we'll end up copy-pasting it all around the place.